internal/services/apimanagement/api_management_policy_fragment_resource_test.go

```diff
@@ -0,0 +1,178 @@
+	})
+}
+
+func (ApiManagementPolicyFragmentResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := policyfragment.ParsePolicyFragmentID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.ApiManagement.PolicyFragmentClient.Get(ctx, *id, policyfragment.DefaultGetOperationOptions())
+	if err != nil {
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	return pointer.To(resp.Model != nil), nil
+}
+
+func (r ApiManagementPolicyFragmentResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_api_management_policy_fragment" "test" {
+  api_management_id = azurerm_api_management.test.id
+  name              = azurerm_api_management.test.name
+  description       = "Some descriptive text"
+  value             = file("testdata/api_management_policy_fragment_test_xml.xml")
+}
+`, r.template(data))
+}
+
+func (r ApiManagementPolicyFragmentResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_api_management_policy_fragment" "import" {
+  api_management_id = azurerm_api_management_policy_fragment.test.api_management_id
+  name              = azurerm_api_management_policy_fragment.test.name
+  description       = azurerm_api_management_policy_fragment.test.description
+  value             = azurerm_api_management_policy_fragment.test.value
+}
+`, r.basic(data))
+}
+
+func (r ApiManagementPolicyFragmentResource) update(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_api_management_policy_fragment" "test" {
+  api_management_id = azurerm_api_management.test.id
+  name              = azurerm_api_management.test.name
+  description       = "Some descriptive text which is updated"
+  value             = file("testdata/api_management_policy_fragment_test_xml.xml")
+}
+`, r.template(data))
+}
+
+func (r ApiManagementPolicyFragmentResource) rawxml(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_api_management_policy_fragment" "test" {
+  api_management_id = azurerm_api_management.test.id
+  name              = azurerm_api_management.test.name
+  description       = "Some descriptive text"
+  format            = "rawxml"
+  value             = file("testdata/api_management_policy_fragment_test_rawxml.xml")
+}
+`, r.template(data))
+}
+
+func (ApiManagementPolicyFragmentResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_api_management" "test" {
+  name                = "acctestAM-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  publisher_name      = "pub1"
+  publisher_email     = "pub1@email.com"
+  sku_name            = "Developer_1"
```

Comment:
Can we update this to use the consumption sku here? It shortens the provisioning time for the APIM resource from 20+ mins to about 2-3 mins which makes the testing iterations quicker/convenient 
```suggestion
  sku_name            = "Consumption_0"
```